[spec/traits.dd] Manually group traits for Table of Contents

ntrel opened the pull request:
This makes it easier to find a particular trait and is better for navigation.

dkorpel reviewed (commented):
I like the categories, though the order of the contents doesn't match the order of the page anymore, and it doesn't have the usual "Contents [hide]" like other spec pages.

ntrel commented:
@dkorpel
> the order of the contents doesn't match the order of the page anymore, 

I'm happy to reorder the main traits items to match the groupings, then I can switch the TOC back to automatic generation. Shall I do that?

> it doesn't have the usual "Contents [hide]"

Now fixed, thanks.

dkorpel commented:
> I'm happy to reorder the main traits items to match the groupings, then I can switch the TOC back to automatic generation. Shall I do that?

That would be nice, yes :)

dkorpel commented on spec/traits.dd:
```diff
@@ -2,7 +2,102 @@ Ddoc
+    $(NAV_SUB_ITEM getOverloads)
+    $(NAV_SUB_ITEM getUnitTests)
+    $(NAV_SUB_ITEM isModule)
+    $(NAV_SUB_ITEM isPackage)
```
I would put `isPackage` and `isModule` under Symbols, not Aggregates

ntrel replied:
OK, now done.

dkorpel commented on spec/traits.dd:
```diff
@@ -2,7 +2,102 @@ Ddoc
+    $(NAV_SUB_ITEM getAliasThis)
+    $(NAV_SUB_ITEM getMember)
+    $(NAV_SUB_ITEM getOverloads)
+    $(NAV_SUB_ITEM getUnitTests)
```
I see the text considers a module an aggregate "Takes one argument, a symbol of an aggregate (e.g. struct/class/module).", I thought an aggregate was a class, struct or union, so I would put this under Symbols.

ntrel replied:
OK, but for consistency I have also moved the following to Symbols as they apply to modules too:
* allMembers
* hasMember
* getOverloads
* getUnitTests
* parent

dkorpel commented on spec/traits.dd:
```diff
@@ -2,7 +2,102 @@ Ddoc
+    $(NAV_SUB_ITEM getCppNamespaces)
+    $(NAV_SUB_ITEM getLinkage)
+    $(NAV_SUB_ITEM getTargetInfo)
+    $(NAV_SUB_ITEM toType)
```
This is not about linkage, but about reification of types

ntrel replied:
OK, moved to Symbols.

ntrel commented:
Rebased and moved traits to just 4 groups (much less items to move, less disruption, easier to review):
* Types (now including some OOP e.g. `isFinalClass`)
* Functions (now including some OOP e.g. `isVirtualFunction`)
* Function Parameters
* Symbols

If further subgroups are desirable, perhaps it would be best to do in a subsequent PR (I think I'm done for now though ;-)).

I spread the changes across small commits to help ease reviewing. No changes to content except one separate commit that fixes a link.
